server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileBackuperImpl.java

```diff
@@ -0,0 +1,215 @@
+import javax.xml.stream.XMLStreamException;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang.builder.CompareToBuilder;
+import org.codehaus.staxmate.SMInputFactory;
+import org.codehaus.staxmate.in.SMHierarchicCursor;
+import org.codehaus.staxmate.in.SMInputCursor;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.api.server.ServerSide;
+import org.sonar.api.utils.text.XmlWriter;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.organization.OrganizationDto;
+import org.sonar.db.qualityprofile.ActiveRuleDto;
+import org.sonar.db.qualityprofile.ActiveRuleParamDto;
+import org.sonar.db.qualityprofile.QualityProfileDto;
+
+@ServerSide
+public class QProfileBackuperImpl implements QProfileBackuper {
+
+  private static final Joiner RULE_KEY_JOINER = Joiner.on(", ").skipNulls();
+
+  private final QProfileReset reset;
+  private final DbClient db;
+
+  public QProfileBackuperImpl(QProfileReset reset, DbClient db) {
+    this.reset = reset;
+    this.db = db;
+  }
+
+  @Override
+  public void backup(DbSession dbSession, QualityProfileDto profileDto, Writer writer) {
+    List<ActiveRuleDto> activeRules = db.activeRuleDao().selectByProfileKey(dbSession, profileDto.getKey());
+    activeRules.sort(BackupActiveRuleComparator.INSTANCE);
+    writeXml(dbSession, writer, profileDto, activeRules.iterator());
+  }
+
+  private void writeXml(DbSession dbSession, Writer writer, QualityProfileDto profile, Iterator<ActiveRuleDto> activeRules) {
+    XmlWriter xml = XmlWriter.of(writer).declaration();
+    xml.begin("profile");
+    xml.prop("name", profile.getName());
+    xml.prop("language", profile.getLanguage());
+    xml.begin("rules");
+    while (activeRules.hasNext()) {
+      ActiveRuleDto activeRule = activeRules.next();
+      xml.begin("rule");
+      xml.prop("repositoryKey", activeRule.getKey().ruleKey().repository());
+      xml.prop("key", activeRule.getKey().ruleKey().rule());
+      xml.prop("priority", activeRule.getSeverityString());
+      xml.begin("parameters");
+      for (ActiveRuleParamDto param : db.activeRuleDao().selectParamsByActiveRuleId(dbSession, activeRule.getId())) {
+        xml
+          .begin("parameter")
+          .prop("key", param.getKey())
+          .prop("value", param.getValue())
+          .end();
+      }
+      xml.end("parameters");
+      xml.end("rule");
+    }
+    xml.end("rules").end("profile").close();
+  }
+
+  @Override
+  public QProfileRestoreSummary restore(DbSession dbSession, Reader backup, OrganizationDto organization, @Nullable String overriddenProfileName) {
+    try {
+      String profileLang = null;
+      String profileName = null;
+      List<RuleActivation> ruleActivations = Lists.newArrayList();
+      SMInputFactory inputFactory = initStax();
+      SMHierarchicCursor rootC = inputFactory.rootElementCursor(backup);
+      rootC.advance(); // <profile>
+      if (!rootC.getLocalName().equals("profile")) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move the "profile" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1132)